mantaray/received.py

```diff
@@ -312,12 +332,27 @@ def _handle_cap(server_view: views.ServerView, args: list[str]) -> None:
     if subcommand == "ACK":
         acknowledged = set(args[-1].split())
         if "sasl" in acknowledged:
+            server_view.core.cap_list.add("sasl")
             server_view.core.send("AUTHENTICATE PLAIN")
+        if "away-notify" in acknowledged:
+            server_view.core.cap_list.add("away-notify")
     elif subcommand == "NAK":
         rejected = set(args[-1].split())
         if "sasl" in rejected:
             # TODO: this good?
             raise ValueError("The server does not support SASL.")
+        if "away-notify" in rejected:
+            raise ValueError("The server does not support away-notify.")
+
+    server_view.core.cap_req.pop()  # To evaluate how many more ACK/NAKs will be received from server
```

Comment:
I don't like how responding to a capability `foo` can cause capability `bar` to be popped from the list. Can you replace `cap_req` with an integer that counts the number of accepted/rejected capabilities?